test/unit/test-chunk.js

```diff
@@ -335,6 +336,100 @@ describe('chunk', () => {
   );
 });
 
+describe('taskQueues', () => {
+  describes.fakeWin(
+    'long chunk tasks force a macro task between work',
+    {
+      amp: true,
+    },
+    env => {
+      let subscriptions;
+      let sandbox;
+      let clock;
+      let progress;
+
+      function complete(str, long) {
+        return function(unusedIdleDeadline) {
+          if (long) {
+            // Ensure this task takes a long time beyond the 5ms buffer.
+            clock.tick(100);
+          }
+          progress += str;
+        };
+      }
+
+      beforeEach(() => {
+        subscriptions = {};
+        sandbox = sinon.sandbox;
+        clock = sandbox.useFakeTimers();
+
+        if (!env.win.addEventListener || !env.win.postMessage) {
+          env.win.addEventListener = function(type, handler) {
+            if (subscriptions[type] && !subscriptions[type].includes(handler)) {
+              subscriptions[type].push(handler);
+            } else {
+              subscriptions[type] = [handler];
+            }
+          };
+          env.win.postMessage = function(key) {
+            if (subscriptions['message']) {
+              subscriptions['message']
+                .slice()
+                .forEach(method => method({data: key}));
+            }
+          };
+        }
+
+        progress = '';
+      });
+
+      afterEach(() => {
+        sandbox.restore();
+      });
+
+      it('should execute chunks after long task in a macro task', done => {
+        startupChunk(env.win.document, complete('a', false));
+        startupChunk(env.win.document, () => {
+          expect(progress).to.equal('a');
+          complete('b', true)();
+          startupChunk(env.win.document, complete('c', false));
```

Comment:
Since `b` took too long and now you're adding another task `c`. 

Should the result on [Line 399](https://github.com/ampproject/amphtml/pull/22532/files#diff-a91c93492b4df52ed589fefd999f8d44R399) be just `ab`?
 
As the `c` should have been shifted to another macro task.